contracts/strategies/StrategyDAI3pool.sol

```diff
@@ -0,0 +1,334 @@
+ - withdraw(uint) - Controller | Vault role - withdraw should always return to vault
+ - withdrawAll() - Controller | Vault role - withdraw should always return to vault
+ - balanceOf()
+ 
+ Where possible, strategies must remain as immutable as possible, instead of updating variables, we update the contract by linking it in the controller
+ 
+*/
+
+contract StrategyDAI3pool {
+    using SafeERC20 for IERC20;
+    using Address for address;
+    using SafeMath for uint256;
+
+    address public constant want = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    address public constant _3pool = address(0xbEbc44782C7dB0a1A60Cb6fe97d0b483032FF1C7);
+    address public constant _3crv = address(0x6c3F90f043a72FA612cbac8115EE7e52BDe6E490);
+    address public constant y3crv = address(0x9cA85572E6A3EbF24dEDd195623F188735A5179f);
+
+    address public constant ypool = address(0x45F783CCE6B7FF23B2ab2D70e416cdb7D6055f51);
+    address public constant ycrv = address(0xdF5e0e81Dff6FAF3A7e52BA697820c5e32D806A8);
+    address public constant yycrv = address(0x5dbcF33D8c2E976c6b560249878e6F1491Bca25c);
+
+    address public constant dai = address(0x6B175474E89094C44Da98b954EedeAC495271d0F);
+    address public constant ydai = address(0x16de59092dAE5CcF4A1E6439D611fd0653f0Bd01);
+    address public constant usdc = address(0xA0b86991c6218b36c1d19D4a2e9Eb0cE3606eB48);
+    address public constant yusdc = address(0xd6aD7a6750A7593E092a9B218d66C0A814a3436e);
+    address public constant usdt = address(0xdAC17F958D2ee523a2206206994597C13D831ec7);
+    address public constant yusdt = address(0x83f798e925BcD4017Eb265844FDDAbb448f1707D);
+    address public constant tusd = address(0x0000000000085d4780B73119b644AE5ecd22b376);
+    address public constant ytusd = address(0x73a052500105205d34Daf004eAb301916DA8190f);
+
+    address public governance;
+    address public controller;
+    address public strategist;
+
+    uint256 public constant DENOMINATOR = 10000;
+    uint256 public treasuryFee = 500;
+    uint256 public withdrawalFee = 0;
+    uint256 public strategistReward = 50;
+    uint256 public threshold = 6000;
+    uint256 public slip = 100;
+    uint256 public tank = 0;
+    uint256 public p = 0;
+    bool public flag = true;
+
+    modifier isAuthorized() {
+        require(msg.sender == strategist || msg.sender == governance || msg.sender == controller || msg.sender == address(this), "!authorized");
+        _;
+    }
+
+    constructor(address _controller) public {
+        governance = msg.sender;
+        strategist = msg.sender;
+        controller = _controller;
+    }
+
+    function getName() external pure returns (string memory) {
+        return "StrategyDAI3pool";
+    }
+
+    function unwindHelper() internal {
+        uint256 _want = IERC20(want).balanceOf(address(this));
+        IERC20(want).safeApprove(ypool, 0);
+        IERC20(want).safeApprove(ypool, _want);
+
+        _want = _want.div(3);
```

Comment:
Just to confirm i understand whats happening, we are calculating 2/3 of the pool from DAI amount to exchange and withdraw into USDT and USDC from ypool and then deposit the three tokens into 3pool, is that correct?

Reply:
correct. because a big amount of DAI is withdrawn from ypool, swapping back atomically and then adding liquidity equally with 3 tokens together could avoid the slippage the most.